src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -950,9 +950,7 @@ static ProcessInfo[] GetProcessInfos(IntPtr dataPtr)
             while (true)
             {
                 IntPtr currentPtr = (IntPtr)((long)dataPtr + totalOffset);
-                SystemProcessInformation pi = new SystemProcessInformation();
-
-                Marshal.PtrToStructure(currentPtr, pi);
+                SystemProcessInformation pi = Marshal.PtrToStructure<SystemProcessInformation>(currentPtr);
```

Comment:
Both SystemProcessInformation and SystemThreadInformation are nice blittable structs. No need to call PtrToStructure (expensive API). You can just cast the pointer to the type: `ref SystemProcessInformation pi = ref *(SystemProcessInformation *)(currentPtr);`